### DISCUSSION TITLE
Convert WinForms format to Avalonia

### DISCUSSION THREAD

**Discussion by AviadBly** [AvaloniaUI/Avalonia#12127] [Q&A]
Hi, How do I convert the following lines to work with avalonia ui format?

List<Form> openForms = new List<Form>(Application.OpenForms.Count);
                openForms = Application.OpenForms.Cast<Form>().ToList();
                openForms.Remove(this);// Removes the main form from the list
                Screen[] screens = Screen.AllScreens;

**Comment by timunie** [comment]
Normally not the avalonia way at all. What are you trying to do?

**Reply by AviadBly** [reply]
Pretty much trying to create a list of the open windows, and also get a list for the screens.
the goal is to spread a number of windows across screens and make them fit and not overlap.

**Reply by timunie** [reply]
okay that is not that easy in Avalonia I guess as not all WindowManagers do allow it. Anyway have a look into ControlCatalog especially the screens page for how it can be accessed. https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog/Pages/ScreenPage.cs
